resource-service/common/gitsuite_test.go

```diff
@@ -992,6 +994,110 @@ func (s *BaseSuite) Test_getGitKeptnEmail(c *C) {
 
 }
 
+func Test_getAuthMethod(t *testing.T) {
+	tests := []struct {
+		name           string
+		gitContext     common_models.GitContext
+		wantErr        bool
+		expectedOutput transport.AuthMethod
+	}{
+		{
+			name: "valid credentials",
+			gitContext: common_models.GitContext{
+				Credentials: &common_models.GitCredentials{
+					RemoteURI:         "https://some.url",
+					Token:             "some-token",
+					User:              "user",
+					GitPrivateKey:     "",
+					GitPrivateKeyPass: "",
+					GitProxyURL:       "",
+					GitProxyInsecure:  false,
+					GitProxyScheme:    "",
+					GitProxyUser:      "",
+					GitProxyPassword:  "",
+				},
+				Project: "my-proj",
+			},
+			wantErr: false,
+			expectedOutput: &http.BasicAuth{
+				Username: "user",
+				Password: "some-token",
+			},
+		},
+		{
+			name: "invalid credentials",
+			gitContext: common_models.GitContext{
+				Credentials: &common_models.GitCredentials{
+					RemoteURI:         "https://some.url",
+					Token:             "",
+					User:              "user",
+					GitPrivateKey:     "",
+					GitPrivateKeyPass: "",
+					GitProxyURL:       "",
+					GitProxyInsecure:  false,
+					GitProxyScheme:    "",
+					GitProxyUser:      "",
+					GitProxyPassword:  "",
+				},
+				Project: "my-proj",
+			},
+			wantErr:        false,
+			expectedOutput: nil,
+		},
+		{
+			name: "invalid ssh credentials",
```

Comment:
please also add a test case for valid SSH credentials

Reply:
that's an integration test